service.go

```diff
@@ -151,6 +152,85 @@ func setupTempLog() *Logs {
 	return &logs
 }
 
+func SwitchLogLevel(logfile string, minLogLevel string) (LogType, error) {
+	// validate log level
+	minLogLevelType, err := validateLogType(minLogLevel)
+	if err != nil {
+		return minLogLevelType, err
+	}
+
+	if Log == nil {
+		//  initialize logger
+		SetupLogWriters(logfile, []io.Writer{}, minLogLevelType)
+		return minLogLevelType, nil
+	}
+
+	var levelWriter = Log.Error.Writer()
+
+	switch minLogLevelType {
```

Comment:
If feel like this function is getting too complicated and `setLoggers` should be unnecessary.

I was thinking you could split up the existing `SetupLogWriters` into two parts: one that does the initial setup, storing a global (private) file writer and extraLogWriters slice, and then that calls `SetLogLevel` which (re)creates all the writers at the levels, very similar to how it is currently doing it, but reusing those global variables.

Does that make sense?

Reply:
Yeah that makes sense. I'll update it to simplify the logic.

Reply:
I've updated to simplify the LogSetup and switch. Gone back to the initial setup and now have function `setLogLevel` which hich (re)creates all the writers at the levels.

`SetupLogWriters` & `SwitchLogLevel` both use `setLogLevel`.

`setLogLevel` uses the global variables `extraLogLevelWriters` & `fileLogWriter` for creating the new/updated log levels.